src/libpmemobj/ulog.h

```diff
@@ -70,15 +70,20 @@ struct ulog_entry_buf {
  * the next field is always allocated with extra padding, and then the offset
  * is additionally aligned.
  */
-#define ULOG(capacity_bytes) {\
-	/* 64 bytes of metadata */\
-	uint64_t checksum; /* checksum of ulog header and its entries */\
-	uint64_t next; /* offset of ulog extension */\
-	uint64_t capacity; /* capacity of this ulog in bytes */\
-	uint64_t gen_num; /* generation counter */\
-	uint64_t unused[4]; /* must be 0 */\
-	uint8_t data[capacity_bytes]; /* N bytes of data */\
-}\
+struct ulog {
+	/* 64 bytes of metadata */
+	uint64_t checksum; /* checksum of ulog header and its entries */
+	uint64_t next; /* offset of ulog extension */
+	uint64_t capacity; /* capacity of this ulog in bytes */
+	uint64_t gen_num; /* generation counter */
+	uint64_t unused[4]; /* must be 0 */
+};
+
+static inline uint8_t *
```

Comment:
add a comment that this data structure must be immediately followed by a buffer of the size for the log.

Reply:
Done.